server/balancer_worker.go

```diff
@@ -102,9 +105,19 @@ func (bw *balancerWorker) addBalanceOperator(regionID uint64, op *BalanceOperato
 		return false
 	}
 
+	// If the region is set balanced some time before, we can't set
+	// it again in a time interval.
+	_, ok = bw.regionCache.get(regionID)
+	if ok {
+		return false
+	}
+
 	// TODO: should we check allowBalance again here?
 
 	bw.balanceOperators[regionID] = op
+
+	bw.regionCache.set(regionID, nil)
```

Comment:
I think that we may add context as value to help our filter strategy, so `set` is more convenient than `add`.